includes/class-starter-content.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * Starter content management class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WP_Error, \WP_Query;
+
+defined( 'ABSPATH' ) || exit;
+
+define( 'NEWSPACK_STARTER_CONTENT_CATEGORIES', '_newspack_starter_content_categories' );
+define( 'NEWSPACK_STARTER_CONTENT_POSTS', '_newspack_starter_content_posts' );
+
+/**
+ * Manages settings for PWA.
+ */
+class Starter_Content {
+
+	/**
+	 * Starter categories;
+	 *
+	 * @var array A set of starter categories.
+	 */
+	protected static $starter_categories = [ 'Featured', 'Sports', 'Entertainment', 'Opinion', 'News', 'Events', 'Longform', 'Arts', 'Politics', 'Science', 'Tech', 'Health' ];
+
+	/**
+	 * Create a single post.
+	 *
+	 * @return int Post ID
+	 */
+	public static function create_post() {
+		if ( ! function_exists( 'wp_insert_post' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+		}
+		$page_templates = [ '', 'single-feature.php', 'single-wide.php' ];
+		$paragraphs = explode( PHP_EOL, self::get_lipsum( 'paras', 5 ) );
+		$title      = self::get_lipsum( 'words', wp_rand( 4, 7 ) );
+		$post_data  = [
+			'post_title'    => $title,
+			'post_name'     => sanitize_title_with_dashes( $title, '', 'save' ),
+			'post_status'   => 'publish',
+			'page_template' => $page_templates[ wp_rand( 0, 2 ) ],
+			'post_content'  => html_entity_decode(
+				implode(
+					'',
+					array_map(
+						function( $paragraph ) {
+							return self::create_block( 'paragraph', null, '<p>' . $paragraph . '</p>' );
+						},
+						$paragraphs
+					)
+				)
+			),
+		];
+
+		$post_id = wp_insert_post( $post_data );
+
+		$newspack_featured_image_positions = [ null, 'behind', 'beside' ];
+
+		$newspack_featured_image_position = $newspack_featured_image_positions[ wp_rand( 0, 2 ) ];
+
+		if ( $newspack_featured_image_position ) {
+			add_post_meta( $post_id, 'newspack_featured_image_position', $newspack_featured_image_position );
+		}
+
+		$attachment_id = self::add_featured_image( $post_id );
+
+		$categories = get_categories(
```

Comment:
There are a few issues with the category-setting logic (see the Count column):

<img width="968" alt="Screen Shot 2019-11-03 at 12 11 32 PM" src="https://user-images.githubusercontent.com/7317227/68089893-250e6b80-fe33-11e9-85f3-d7573f7e19fb.png">


I think a much simpler way of finding the correct category to set would be like:

```
$categories = get_categories([
	'orderby'    => 'count',
	'order'      => 'ASC',
	'hide_empty' => false,
	'exclude'    => '1', // Exclude 'Uncategorized' which has ID of 1.
	'number'     => 1,
]);
$category_to_set = $categories[0];
```
This should get the category containing the least amount of posts, so it will always evenly distribute the posts among the categories.

Reply:
Good call - this is much more elegant. Change made  in https://github.com/Automattic/newspack-plugin/pull/299/commits/e631432bfa9966033fc981500088d98843938f37